linalg/densemat.cpp

```diff
@@ -3063,6 +3077,53 @@ void Add(double alpha, const DenseMatrix &A,
    Add(alpha, A.GetData(), beta, B.GetData(), C);
 }
 
+bool LinearSolve(DenseMatrix& A, double* X, double TOL)
+{
+   MFEM_VERIFY(A.IsSquare(), "A must be a square matrix!");
+   MFEM_ASSERT(A.NumCols() > 0, "supplied matrix, A, is empty!");
+   MFEM_ASSERT(X != nullptr, "supplied vector, X, is null!");
+
+   int N = A.NumCols();
+
+   switch (N)
+   {
+      case 1:
+      {
+         double det = A(0,0);
+         if (std::abs(det) <= TOL) { return false; } // singular
+
+         X[0] /= det;
+         break;
+      }
+      case 2:
+      {
+         double det = A.Det();
+         if (std::abs(det) <= TOL) { return false; } // singular
+
+         double invdet = 1. / det;
+
+         double b0 = X[0];
+         double b1 = X[1];
+
+         X[0] = ( A(1,1)*b0 - A(0,1)*b1) * invdet;
+         X[1] = (-A(1,0)*b0 + A(0,0)*b1) * invdet;
+         break;
+      }
+      default:
+      {
+         // default to LU factorization for the general case
+         Array<int> ipiv(N);
+         LUFactors lu(A.Data(), ipiv);
+
+         if (!lu.Factor(N)) { return false; } // singular
```

Comment:
Do you want `lu.Factor(N)` or `lu.Factor(N,TOL)` here?

Reply:
Good catch! My intent was to definitely propagate the specified tolerance, `TOL` to the LU factorization call, `lu.Factor()`. Fixed.